File: app/src/main/java/com/delarosa/cognitoAuth/data/AuthUtils.java
package com.delarosa.cognitoAuth.data;

import android.net.Uri;
import android.util.Log;

import com.amazonaws.mobileconnectors.cognitoauth.Auth;

public class AuthUtils {
    private static Auth auth;

    /**
     * Handles button press.
     *
     * @param signIn When {@code True} this performs sign-in.
     */
    public static void onButtonPress(boolean signIn) {
        Log.d(" -- ", "Button press: " + signIn);
        if (signIn) {
            auth.getSession();
        } else {
            auth.signOut();
        }
    }

    /**
     * guarda la sesion
     * @param builder
     */
    public static void buildAuth(Auth.Builder builder) {

        auth = builder.build();

    }

    /**
     * obtiene los tokens
     * @param data
     */
    public static void getTokens(Uri data) {
        auth.getTokens(data);
    }
}

File: app/src/main/java/com/delarosa/cognitoAuth/model/AuthUtils.kt
package com.delarosa.cognitoAuth.model

import android.annotation.SuppressLint
import android.net.Uri
import android.util.Log

import com.amazonaws.mobileconnectors.cognitoauth.Auth

object AuthUtils {
     @SuppressLint("StaticFieldLeak")
     var auth: Auth? = null

    /**
     * guarda la sesion
     * @param builder
     */
    fun buildAuth(builder: Auth.Builder) {
        auth = builder.build()
    }

    /**
     * gestiona el boton de inicio o cierre de sesion
     * @param signIn When `True` this performs sign-in.
     */
    fun onButtonPress(signIn: Boolean) {
        Log.d(" -- ", "Button press: $signIn")
        if (signIn) {
            auth!!.getSession()
        } else {
            auth!!.signOut()
        }
    }

    /**
     * obtiene los tokens
     * @param data
     */
    fun getTokens(data: Uri) {
        auth!!.getTokens(data)
    }

}
